### PR TITLE
promoting "NodeConformance" tagged tests in securityContext tests for conformance suite - step 1 : refactor

### PR DESCRIPTION
**Pull request by mkurylec**
These are the promoted testcases:

- test/e2e_node/security_context_test.go: "should run the container with uid 65534"
- test/e2e_node/security_context_test.go: "should run the container with uid 0"
- test/e2e_node/security_context_test.go: "should run the container with readonly rootfs when readOnlyRootFilesystem=true"
- test/e2e_node/security_context_test.go: "should run the container with writable rootfs when readOnlyRootFilesystem=false"
- test/e2e_node/security_context_test.go: "should allow privilege escalation when not explicitly set and uid != 0"
- test/e2e_node/security_context_test.go: "should not allow privilege escalation when false"
- test/e2e_node/security_context_test.go: "should allow privilege escalation when true"

[This is the Umbrella Issue](https://github.com/kubernetes/kubernetes/issues/66875) that consolidates the effort.

### CODE REVIEW

**Comment by neolit123** [comment]
/sig testing
/ok-to-test

/release-note-none
^ ?

**Comment by spiffxp** [comment]
/sig node
/hold
So these need to be ported to test/e2e/common if these features should be considered part of conformance.  It's not immediately clear to me, will need help.  Near as I can tell none of them check things in a way that they would have to be run on a node, so I wouldn't see anything wrong with these getting ported to e2e/common.  @yujuhong @wangzhen127 any thoughts here?

The "should allow privilege escalation" tests are being gated by whether `ContainerRuntime` is "docker", @filbranden is this still necessary?

**Comment by wangzhen127** [comment]
The promoted Node e2e tests should live in test/e2e/common. I think the security context tests should be fine getting ported. The guideline is that those ported tests should not use ssh, as least.

**Comment by filbranden** [comment]
@spiffxp The "AllowPrivilegeEscalation" e2e tests are not really gated on ContainerRuntime being "docker"... That block is simply doing additional checks to see if Docker supports "NoNewPrivileges", which at some point was broken on RHEL/CentOS 7 due to SELinux policy.

The issue with SELinux policy (https://github.com/projectatomic/container-selinux/issues/45) is now marked as Closed, so I think it's quite probably OK to drop that block.

If you'd like to do some testing yourself, take a look at my commit 6d30b026baed6e1906f96e7b982e7c205be7430d for instructions of how I tested that first. (Though I wouldn't be sad if you remove that without retesting it on that platform... It's fine.)

Cheers,
Filipe

**Comment by spiffxp** [comment]
```
W0813 15:43:30.180] Run ./hack/update-gofmt.sh
```

**Comment by smarterclayton** [comment]
In general any changes to tests need to be done in a separate PR, then the actual promotion to conformance is supposed to happen in its own PR for tracking and approval.

**Comment by bgrant0607** [comment]
How portable are these tests to different Linux OSes and CRI implementations? 

Windows is a bigger ball of wax -- we'll likely need to recategorize quite a few tests, rebuild test images, etc. in order to make that work. So I'm not concerned about that for now.

**Comment by spiffxp** [comment]
/approve
/cc @wangzhen127 
For the /lgtm

Is there anything holding the current PR back from being merged before we get into whether it should be promoted to conformance?

File: test/e2e_node/security_context_test.go [hunk 1]
```diff
@@ -364,77 +364,50 @@ var _ = framework.KubeDescribe("Security Context", func() {
...
-							Image:   image,
-							Name:    podName,
-							Command: command,
-							SecurityContext: &v1.SecurityContext{
-								ReadOnlyRootFilesystem: &readOnlyRootFilesystem,
-							},
-						},
-					},
-				},
+		It("should run the container as unprivileged when false [NodeConformance]", func() {
```
Comment (wangzhen127) [inline test/e2e_node/security_context_test.go]: Is this PR moving all [NodeConformance] tests to e2e/common? Should this and other remaining [NodeConformance] tests be moved?

**Comment by spiffxp** [comment]
/assign @mgdevstack 
can you take a look / open a new PR if necessary?

**Comment by mgdevstack** [comment]
@spiffxp, seems PR doesn't require any updates at this moment. Once @wangzhen127 is good with review and [comment](https://github.com/kubernetes/kubernetes/pull/67032#discussion_r212248409), PR is good to get merged.

File: test/e2e_node/security_context_test.go (hunk 1 above)
Comment (wangzhen127) [reply test/e2e_node/security_context_test.go]: Given the [comments from Filipe](https://github.com/kubernetes/kubernetes/pull/67032#issuecomment-411187287), it seems to me that we can remove the container runtime constraint.

And why is privileged mode a problem?

**Comment by spiffxp** [comment]
>it seems to me that we can remove the container runtime constraint.

@mgdevstack this the additional work I think this PR needs

**Comment by wangzhen127** [comment]
> And why is privileged mode a problem?

And also this. :)

File: test/e2e_node/security_context_test.go (hunk 1 above)
Comment (mgdevstack) [reply test/e2e_node/security_context_test.go]: >Given the comments from Filipe, it seems to me that we can remove the container runtime constraint.

Is there any chances that few k8s workloads might be running on CentOS < 7 or running with the lower container-selinux version where the [mentioned patch](https://github.com/containers/container-selinux/issues/45) is not added?

- If this is not a concern, I can create new PR to add [remaining [NodeConformance] tests](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/security_context_test.go#L500-L521) validating logs under AllowPrivilegeEscalation PodSpec. Removing [ContainerRuntime constraint](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/security_context_test.go#L443-L464).

>And why is privileged mode a problem?

Privileged mode is not a problem. It's my bad, I missed to notice.
- New PR would include [remaining e2e tests](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/security_context_test.go#L558-L582), validating logs under privileged PodSpec. Removing [[NodeFeature:HostAccess]](https://github.com/kubernetes/kubernetes/blob/55859a60fe09c412e183c92ad265cf0d52fbe3d8/test/e2e_node/security_context_test.go#L558:56-65) from test name

Comment (wangzhen127) [reply test/e2e_node/security_context_test.go]: > Is there any chances that few k8s workloads might be running on CentOS < 7 or running with the lower container-selinux version where the mentioned patch is not added?
> - If this is not a concern, I can create new PR to add remaining [NodeConformance] tests validating logs under AllowPrivilegeEscalation PodSpec. Removing ContainerRuntime constraint.

+@filbranden

> Privileged mode is not a problem. It's my bad, I missed to notice.
> - New PR would include remaining e2e tests, validating logs under privileged PodSpec. Removing [NodeFeature:HostAccess] from test name

Do you mean you will move those in a separate PR? Only the tests marked with [NodeConformance] should be moved. Let's discuss in the followup PR on whether we should change [NodeFeature:HostAccess] to [NodeConformance].

**Comment by wangzhen127** [comment]
/lgtm
I am fine with fixing the 2 remaining points in followup PRs.

File: test/e2e_node/security_context_test.go (hunk 1 above)
Comment (filbranden) [reply test/e2e_node/security_context_test.go]: > Is there any chances that few k8s workloads might be running on CentOS < 7 or running with the lower container-selinux version where the mentioned patch is not added?
> * If this is not a concern, I can create new PR to add remaining [NodeConformance] tests validating logs under AllowPrivilegeEscalation PodSpec. Removing ContainerRuntime constraint.

Fine to proceed with that.

If tests are run on CentOS, that should be on CentOS 7 with all the latest packages and that includes an appropriate container-selinux version.

**Comment by spiffxp** [comment]
```
__main__/test/e2e/common/security_context.go:62:5: undefined: busyboxImage
```
I think this needs to be rebased @mgdevstack may be easier to open a new PR

**Pull request closed without merging**